sycl/test/check_device_code/hip/atomic/amdgpu_unsafe_atomics.cpp

```diff
@@ -27,16 +26,14 @@ SYCL_EXTERNAL void fpAtomicFunc(float *f, double *d) {
       .fetch_add(1.0f);
   // CHECK: void{{.*}}fpAtomicFunc
   // CHECK-SAFE: atomicrmw volatile fadd
-  // CHECK-SAFE-NOT: llvm.amdgcn.global.atomic.fadd.f32
-  // CHECK-UNSAFE-FP: llvm.amdgcn.global.atomic.fadd.f32
-  // CHECK-UNSAFE-FP-NOT: atomicrmw volatile fadd
+  // CHECK-SAFE-NOT: amdgpu.ignore.denormal.mode
+  // CHECK-UNSAFE-FP: atomicrmw volatile fadd {{.*}}amdgpu.no.fine.grained.memory{{.*}}amdgpu.ignore.denormal.mode
```

Comment:
Checking the `!` might help make it clear we're checking for metadata.

Alternatively, we could maybe just check the whole line, without skipping anything (`{{.*}}`)? I take it we can't use `update_cc_test_checks.py` to autogenerate everything?

Reply:
I've added the `!`, seems like update tests script is having a hard time trying to parse the output of the tests:
```
!93 = !{!13, !32}
Traceback (most recent call last):
  File "./llvm/utils/update_cc_test_checks.py", line 561, in <module>
    sys.exit(main())
  File "./llvm/utils/update_cc_test_checks.py", line 392, in main
    for k, v in get_line2func_list(ti.args, clang_args).items():
  File "./llvm/utils/update_cc_test_checks.py", line 127, in get_line2func_list
    ast = json.loads(stdout)
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 342, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 23586733 column 2 (char 1320540514)
```
In any case not sure if it would bring that much.

Reply:
Aw, fair enough. Just that automating the tests is like clang-format (don't have to think about it) and means we'd catch any unwanted change to the output, like if the alignment was wrong, or an update brought in extra metadata that we don't expect.

We should probably look at fixing that script for SYCL (if that is indeed the problem) - I think it would help the upstreaming effort.

Reply:
I've added it to the internal issue tracker.